src/uwtools/config/formats/base.py

```diff
@@ -163,15 +163,20 @@ def dereference(self) -> None:
         """
         Render as much Jinja2 syntax as possible.
         """
-        log.debug("Dereferencing, initial value: %s", self.data)
+
+        def logstate(state: str) -> None:
+            log.debug("Dereferencing, %s value:", state)
+            for line in str(self).split("\n"):
+                log.debug("  %s", line)
```

Comment:
The change from `self.data` to `str(self)` is due to the fact that a `YAMLConfig` may now contain `TaggedString` nodes, whose default string representation is a mess. Calling `str()` on the object invokes its `__repr__()` method, which now registers a "representer" function with the YAML library to neatly serialize a `TaggedString` node, e.g. as `!int "88"`.